src/charm.py

```diff
@@ -649,45 +647,19 @@ def update_status(self):
         try:
             if self.check_pgb_running():
                 self.unit.status = ActiveStatus()
+            else:
+                # Try to restart
+                self.render_pgb_config(True)
         except PebbleConnectionError:
             not_running = "pgbouncer not running"
             logger.error(not_running)
             self.unit.status = WaitingStatus(not_running)
 
-    def _on_upgrade_charm(self, _) -> None:
-        """Re-render the auth file, which is lost in a pod reschedule."""
-        if auth_file := self.get_secret(APP_SCOPE, AUTH_FILE_DATABAG_KEY):
-            self.render_auth_file(auth_file)
-
-    def reload_pgbouncer(self, restart: bool = False) -> None:
-        """Reloads pgbouncer application.
-
-        Pgbouncer will not apply configuration changes without reloading, so this must be called
-        after each time config files are changed.
-
-        Raises:
-            ops.pebble.ConnectionError if pgb service isn't accessible
-        """
-        logger.info("reloading pgbouncer application")
-
-        pgb_container = self.unit.get_container(PGB)
-        pebble_services = pgb_container.get_services()
-        for service in self._services:
-            if service["name"] not in pebble_services:
-                # pebble_ready event hasn't fired so pgbouncer has not been added to pebble config
-                raise PebbleConnectionError
-            if restart or pebble_services[service["name"]].current != ServiceStatus.ACTIVE:
-                pgb_container.restart(service["name"])
-            else:
-                pgb_container.send_signal(SIGHUP, service["name"])
-
-        self.check_pgb_running()
-
     def _generate_monitoring_service(self, enabled: bool = True) -> dict[str, str]:
         if enabled and (stats_password := self.get_secret(APP_SCOPE, MONITORING_PASSWORD_KEY)):
             command = (
                 f'pgbouncer_exporter --web.listen-address=:{METRICS_PORT} --pgBouncer.connectionString="'
-                f'postgres://{self.backend.stats_user}:{stats_password}@localhost:{self.config.listen_port}/pgbouncer?sslmode=disable"'
+                f'postgresql://{self.backend.stats_user}:{stats_password}@localhost:{self.config.listen_port}/pgbouncer?sslmode=disable"'
```

Comment:
It surprises me here. Any comments why it is necessary in this PR?

Reply:
It's not necessary, just to keep it consistent with other uris.